### DISCUSSION TITLE
Sub-Threads/Channels and Advanced Search filters and TAGS for better Productivity and Organization is:issue is:open

### DISCUSSION THREAD

**Discussion by jittygitty** [element-hq/element-meta#1049] [Ideas]
#### What would you like to do?
Be able to Right Click on a channel/room and add a Sub-room/channel/thread underneath that with its own separate Topic and/or "Labels/Tags". 

(Also be able to have advanced filtering and searching, sometimes I want to search for a word from specific user, let's say myself even, and not search through everyone's message, or search through only TAGGED messages etc etc. This Search/Filtering part likely should be in its own separate issue/feature request.)

#### Why would you like to do it?
Element Chat is currently ok just for "chat". I'd like to use it for PRODUCTIVITY, and for that, sadly, in its current form it is very inadequate. I searched and best I could find was ZULIP which seems to support subchannels and tags etc, but I'd rather Element add such features for better organization of chats and productivity usage.

#### How would you like to achieve it?
How would you like to achieve it is a very broad question, how to actually code those features? Who should do it? Or the underlying logic/pseudo-code or UX/UI interface? It might help to break down this question into more specific parts.

I know we currently have BETA "REPLY in THREAD", which unfortunately just isn't implemented well enough UI/UX wise at least to be useful for me. It's not a bad feature, especially if the UX/UI gets better for it, but it does NOT replace the request for having "MULTIPLE" "DIRECT CHATS" with the "SAME" person as Sub-Channels and for different types of discussions, personal, work/biz, and or per-project etc etc.

Ideally, TAGS/LABELS should be the real method used for organization since its very flexible, just like email providers moved from "folder/subfolder" idea to labels/tags where the "hierarchy" can be easily changed at any time, unlike with actual 'folders'.

(In other words, it is 'possible' that the server could be completely "agnostic" to this organization system and clients can send recommendations to the other clients, or others can re-organize the channels in a different way, the way 'they' see fit etc.)

Bottom line, we should be able to simply Right-Click on an existing Direct-Message chat, or a Group-Channel/Room and simply create a new "sub-room" and TAG it etc.

Though I post this request on Element-Web, I am using Element-Desktop mostly.

### Have you considered any alternatives?

Yes, I have considered moving to ZULIP or other similar competing solutions, but would prefer staying with Element due to its federated nature, if such features can be added.

### Additional context

Perhaps the project maintainers can at least provide guidance on implementation in a way that would later be accepted in a pull request, if they have no interest in such features themselves. (Especially how much should go client-only and how much server.)
